### PR TITLE
Support for config files

### PR DESCRIPTION
**Pull request by twenzel**
Test assemblies will be loaded into separate AppDomain.
Minor tweaks:
 - Exception handling
 - Error on markdown file generation if output directory does not exist

### CODE REVIEW

**Comment by Aaronontheweb** [comment]
@twenzel I haven't looked at the code yet, but looking at the build log on TeamCity (which is publicly accessible - click on the red `x` and then click on the perf-tests details link) all of the NBench assertions  pass, but it might be possible that the exit code wasn't correctly reported back to the `NBench.Runner.exe` application.

File: tests/NBench.Tests.End2End/NBenchIntregrationTest.cs [hunk 1]
```diff
@@ -1,40 +1,54 @@
...
+        [Fact]
+        public void LoadAssemblyCorrect()
+        {
+#if DEBUG
+	var package = new TestPackage(@"..\..\..\NBench.Tests.Assembly\bin\Debug\NBench.Tests.Assembly.dll");
+#else
+    var package = new TestPackage(@"..\..\..\NBench.Tests.Assembly\bin\Release\NBench.Tests.Assembly.dll"); 
+#endif
+
+			package.Validate();
```
Comment (Aaronontheweb) [inline tests/NBench.Tests.End2End/NBenchIntregrationTest.cs]: will this test also explicitly check to see if we can load a `.config` file too, or do we need to test that separately?

**Comment by Aaronontheweb** [comment]
looks good, but some comments:

1. You need to squash your commits - I just added [contributing.md](https://github.com/petabridge/NBench/blob/dev/contributing.md) a second ago and that explains the process for doing it.
2. Could you add an explicit unit test that verifies that the external configuration file was loaded? Best way to do that would probably be to modify the `EndToEnd` assembly with a .config file that has some `AppSetting` in it that we can use inside an NBench spec or something. Primary reason for adding this test is regression testing - so no one breaks this feature in the future.

File: tests/NBench.Tests.End2End/NBenchIntregrationTest.cs (hunk 1 above)
Comment (twenzel) [inline tests/NBench.Tests.End2End/NBenchIntregrationTest.cs]: Yes with will break if the config of ```NBench.Tests.Assembly.dll```` can't be read.

**Comment by Aaronontheweb** [comment]
@twenzel why was this closed?

**Pull request closed without merging**